targets/TARGET_NORDIC/TARGET_NRF5/serial_api.c

```diff
@@ -544,9 +544,9 @@ static void internal_set_hwfc(FlowControl type,
                              PinName rxflow, PinName txflow)
 {
     UART_CB.pselrts =
-        ((rxflow == NC) || (type == FlowControlCTS)) ? NRF_UART_PSEL_DISCONNECTED : (uint32_t)rxflow;
+        ((rxflow == NC) || (type == FlowControlCTS)) ? UART_DEFAULT_RTS : (uint32_t)rxflow;
```

Comment:
With NRF51 devices the serial comm gets stopped unless we keep the PIN numbers assigned for RTS/CTS lines. Note that the Flowcontrol is still disabled, its just that PSELRTS, PSELCTS registers need to be programmed.